waffle-chai/src/matchers/emit.ts

```diff
@@ -1,18 +1,23 @@
-import {Contract} from 'ethers';
+import {Contract, providers, utils} from 'ethers';
 
 export function supportEmit(Assertion: Chai.AssertionStatic) {
-  const filterLogsWithTopics = (logs: any[], topic: any, contractAddress: string) =>
+  const filterLogsWithTopics = (logs: providers.Log[], topic: any, contractAddress: string) =>
     logs.filter((log) => log.topics.includes(topic))
       .filter((log) => log.address && log.address.toLowerCase() === contractAddress.toLowerCase());
 
   Assertion.addMethod('emit', function (this: any, contract: Contract, eventName: string) {
     const promise = this._obj;
     const derivedPromise = promise.then((tx: any) =>
       contract.provider.getTransactionReceipt(tx.hash)
-    ).then((receipt: any) => {
-      const eventDescription = contract.interface.events[eventName];
+    ).then((receipt: providers.TransactionReceipt) => {
+      let eventFragment: utils.EventFragment | undefined;
+      try {
+        eventFragment = contract.interface.getEvent(eventName);
+      } catch (e) {
+        // ignore error
```

Comment:
Good rule of the tumb is to rather throw an exception than let it silently fail.